shared/src/test/scala/coop/rchain/shared/MetricsTestImpl.scala

```diff
@@ -0,0 +1,45 @@
+package coop.rchain.shared
+import cats.effect.Sync
+import coop.rchain.metrics.Metrics
+
+import scala.collection.mutable.{Map => MutableMap}
+
+class MetricsTestImpl[F[_]: Sync] extends Metrics[F] {
+  val counters: MutableMap[String, Long] = MutableMap.empty
+  val samplers: MutableMap[String, Long] = MutableMap.empty
+  val gauges: MutableMap[String, Long]   = MutableMap.empty
+  final case class Record(value: Long, count: Long)
+  val records: MutableMap[String, List[Record]] = MutableMap.empty
+
+  private def incrementBy(name: String, delta: Long)(m: MutableMap[String, Long]): Unit = {
+    val newValue = m.get(name).map(_ + delta).getOrElse(delta)
+    println(s"Adding $delta at $name")
+    m.update(name, newValue)
+  }
+
+  private def set[A](name: String, value: A)(m: MutableMap[String, A]): Unit =
+    m.update(name, value)
+
+  override def incrementCounter(name: String, delta: Long): F[Unit] =
+    Sync[F].delay(incrementBy(name, delta)(counters))
+
+  override def incrementSampler(name: String, delta: Long): F[Unit] =
+    Sync[F].delay(incrementBy(name, delta)(samplers))
+
+  // no idea how to implement this properly
+  override def sample(name: String): F[Unit] =
```

Comment:
Maybe leave as `???` if implementation not known.

Reply:
Good idea.